autoload/ale/engine.vim

```diff
@@ -201,6 +201,16 @@ function! s:FixLocList(buffer, loclist) abort
     let l:last_line_number = ale#util#GetLineCount(a:buffer)
 
     for l:item in a:loclist
+        if g:ale_experimental_multibuffer
+            let l:bufnr = a:buffer
+
+            if has_key(l:item, 'bufnr') && l:item.bufnr >= 0
+              let l:bufnr = l:item.bufnr
+            endif
```

Comment:
We could consider changing the API so `bufnr` can be excluded, and then write this:

```vim
if !has_key(l:item, 'bufnr')
    let l:item.bufnr = a:buffer
endif
```